tcl/is.tcl

```diff
@@ -534,10 +266,58 @@ namespace eval qc::is {
         return [expr {[string toupper $bool] in {Y N YES NO TRUE FALSE T F 0 1}}]
     }
 
-    proc decimal {number} {
```

Comment:
Something like 1.5e10 will be cast to 15000000000 by postgres.

So while the string "1.5e10" is not a valid decimal it is still castable so not sure if this proc should return o or 1 in this case ?